src/hooks/plan_tree_walker.c

```diff
@@ -63,15 +72,15 @@ bool plan_tree_walker(Plan *plan, bool (*walker_func)(Node *plan, void *context)
             if(base_plan_walker(&(ctescan->scan.plan), walker_func, context)) return true;
             break;
         }
-        // Join nodes
+#if PG_VERSION_NUM < 160000
         case T_Join:
```

Comment:
what hapened to T_Join in postgres 16?

Reply:
From what I could tell from playing around with EXPLAINs, it stopped being used by Postgres 15. JOINs were handled with another node (I think T_Group?) in Postgres 15. So I think they just dropped it because not being used.